cumulus/parachains/integration-tests/emulated/tests/people/people-rococo/src/tests/reap_identity.rs

```diff
@@ -0,0 +1,305 @@
+			]
+		);
+
+		let reserved_bal = RococoBalances::reserved_balance(RococoRelaySender::get());
+		total_deposit = SubAccountDeposit::get() + id_deposit_relaychain(&id.relay);
+
+		// The reserved balance should equal the calculated total deposit
+		assert_eq!(reserved_bal, total_deposit);
+	});
+	total_deposit
+}
+
+fn assert_set_id_parachain(id: Identity) {
+	// Set identity and Subs on Parachain with Zero deposit
+	PeopleRococo::execute_with(|| {
+		let free_bal = PeopleRococoBalances::free_balance(PeopleRococoSender::get());
+		let reserved_bal = PeopleRococoBalances::reserved_balance(PeopleRococoSender::get());
+
+		//total balance at Genesis should be zero
+		assert_eq!(reserved_bal + free_bal, 0);
+
+		// 3. Set identity on Parachain
+		assert_ok!(PeopleRococoIdentity::set_identity_no_deposit(
+			&PeopleRococoSender::get(),
+			id.parachain.clone(),
+		));
+
+		// 4. Set sub-identity on Parachain
+		assert_ok!(PeopleRococoIdentity::set_sub_no_deposit(
+			&PeopleRococoSender::get(),
+			PeopleRococoReceiver::get(),
+		));
+		// No events get triggered when calling set_sub_no_deposit
+
+		// No amount should be reserved as deposit amounts are set to 0.
+		let reserved_bal = PeopleRococoBalances::reserved_balance(PeopleRococoSender::get());
+		assert_eq!(reserved_bal, 0);
+		assert!(PeopleRococoIdentity::identity(&PeopleRococoSender::get()).is_some());
+		let (_, sub_accounts) =
+			<PeopleRococo as PeopleRococoPallet>::Identity::subs_of(&PeopleRococoSender::get());
+		assert!(sub_accounts.len() > 0);
+	});
+}
+
+fn assert_reap_id_relay(total_deposit: u128) {
+	// 5. reap_identity on Relay Chain
+	RococoRelay::execute_with(|| {
+		type RuntimeEvent = <RococoRelay as Chain>::RuntimeEvent;
+		let free_bal_before_reap = RococoBalances::free_balance(RococoRelaySender::get());
+		let reserved_balance = RococoBalances::reserved_balance(RococoRelaySender::get());
+		//before reap reserved balance should be equal to total deposit
+		assert_eq!(reserved_balance, total_deposit);
+		assert_ok!(RococoIdentityMigrator::reap_identity(
+			RococoOrigin::root(),
+			RococoRelaySender::get(),
+		));
+		assert_expected_events!(
+			RococoRelay,
+			vec![
+				RuntimeEvent::Balances(BalancesEvent::Unreserved { who, amount }) => {
+					who: *who == RococoRelaySender::get(),
+					amount: *amount == total_deposit,
+				},
+			]
+		);
+		assert!(PeopleRococoIdentity::identity(&RococoRelaySender::get()).is_none());
+		let (_, sub_accounts) = RococoIdentity::subs_of(&RococoRelaySender::get());
+		assert_eq!(sub_accounts.len(), 0);
+
+		let reserved_balance = RococoBalances::reserved_balance(RococoRelaySender::get());
+		// after reap reserved balance should be 0
+		assert_eq!(reserved_balance, 0);
+		let free_bal_after_reap = RococoBalances::free_balance(RococoRelaySender::get());
+
+		// free balance should be greater than before reap
+		assert!(free_bal_after_reap > free_bal_before_reap);
+		assert_eq!(free_bal_after_reap, free_bal_before_reap + total_deposit);
```

Comment:
Is this true? Some of the `total_deposit` amount should have been teleported to the parachain.

Reply:
This amount is being transferred after calling `reap_identity`(for the minimal Identity case)
```
RuntimeEvent::Balances(
    RuntimeEvent::Balances(
        Event::Deposit {
            who: ...,
            amount: 1348599900,
        },
    ),
```

So an amount is being transferred, but nothing from `total_deposit` as that value is calculated by functions in this test and don't get mutated after reaping. The assertion here is true yes. 

For the first identity instance, the calculated deposits are:
```
id_deposit_relay = 33959993400
id_deposit_para = 672966597
```

How is the Deposit amount `1348599900` being calculated? I've tried numerous ways to derive the value from the para and relay id deposits but can't find a relationship. I'll need this in order to assert on the inner enum value and check the correct amount is being transferred. It != `total_deposit`.

In other words this assertion will fail once https://github.com/paritytech/polkadot-sdk/issues/2460 is fixed 
```rust
assert_expected_events!(
			RococoRelay,
			vec![
				RuntimeEvent::Balances(BalancesEvent::Unreserved { who, amount }) => {
					who: *who == RococoRelaySender::get(),
					amount: *amount == total_deposit,
				},
			]
		);
```

Reply:
> How is the Deposit amount 1348599900 being calculated? I've tried numerous ways to derive the value from the para and relay id deposits but can't find a relationship. I'll need this in order to assert on the inner enum value and check the correct amount is being transferred. It != total_deposit.

I know how it's being calculated, that's why I'm very surprised that this assertion works.

> In other words, currently, this assertion will fail once https://github.com/paritytech/polkadot-sdk/issues/2460 is fixed

```rust
RuntimeEvent::Balances(BalancesEvent::Unreserved { who, amount }) => {
	who: *who == RococoRelaySender::get(),
	amount: *amount == total_deposit,
},
```
This one, on the other hand, makes complete sense. Why would this fail?

Reply:
I just meant that `total_deposit = 40626660000` whereas `amount = 1348599900`. The `assert_expected_events` macro is giving us a false positive, it should be failing.

Reply:
Can confirm the actual amount == deposit now after this fix https://github.com/paritytech/polkadot-sdk/pull/2552